File: 03_flutter_bank_base_with_ui/FlutterBankCompose/app/src/main/java/kr/flutterbank/aos/compose/AccountManager.kt
package kr.flutterbank.aos.compose

import androidx.compose.runtime.mutableStateListOf

object AccountManager {
  val accounts = mutableStateListOf<Account>()

  init {
    accounts.addAll(
      arrayListOf(Account("10011"), Account("20012"))
    )
  }

  fun checkAccountNoValid(accountNo: String): Boolean {
    return accountNo.isNotBlank() && !accounts.map { it.no }
      .contains(accountNo)
  }

  fun makeAccount(accountNo: String): Boolean {
    accounts.add(Account(accountNo))
    return true
  }
}

File: 03_flutter_bank_base_with_ui/FlutterBankSwiftUI/FlutterBankSwiftUI/AccountManager.swift
import SwiftUI

class AccountManager: ObservableObject {
  static let shared = AccountManager()
  
  @Published var accounts: [Account] = [
    Account(no: "10011"),
    Account(no: "20012")
  ]
  
  func checkAccountNoValid(_ accountNo: String) -> Bool {
    if !accountNo.isEmpty && !accounts.map({ $0.no }).contains(accountNo) {
      return true
    }
    return false
  }
  
  func addAccount(_ accountNo: String) -> Bool {
    accounts.append(Account(no: accountNo))    
    return true
  }
}

